lib/dialects/sqlite3/schema/internal/tokenizer.js

```diff
@@ -0,0 +1,38 @@
+function tokenize(text, tokens) {
+  const compiledRegex = new RegExp(
+    Object.entries(tokens)
+      .map(([type, regex]) => `(?<${type}>${regex.source})`)
+      .join('|'),
+    'yi'
+  );
+
+  let index = 0;
+  const ast = [];
+
+  while (index < text.length) {
+    compiledRegex.lastIndex = index;
+    const result = text.match(compiledRegex);
+
+    if (result !== null) {
+      const [type, text] = Object.entries(result.groups).find(
+        ([_, t]) => t !== undefined
```

Comment:
is `t` `text` or `token` here?

Reply:
`t` is `text` in this case, analogous to `[type, text]`. A token is the whole thing consisting of a type and the text.

Reply:
then let's rename this one to `textEntry` or something along these lines, so that it is less confusing

Reply:
I renamed the destructured array to `[name, group]` as those entries are essentially regex named capture groups and their content.